### PR TITLE
PR #182 split - info.description template review (part 3 of 3)

### LINKED ISSUE #178: The wording in the Mandatory template for info.description is unclear
**Problem description**

The wording in the Mandatory template for info.description is unclear.

```
Which specific authorization flows are to be used will be determined during onboarding process, happening between the API Client and the Telco Operator exposing the API, taking into account the declared purpose for accessing the API, while also being subject to the prevailing legal framework dictated by local legislation.
```

**Expected behavior**
At a minimum add a `the` before onboarding process or use the plural `onboarding processes` if that was the intent.

```diff
- during onboarding process
+ during the onboarding process
```

**Alternative solution**

Alternately, split this into two sentences.

"The specific authorization flows to be used will be determined during the onboarding process between the API client and the Telco operator exposing the API. This determination will consider the declared purpose for accessing the API and be subject to the prevailing legal framework dictated by local legislation."

**Additional context**

We've added this Mandatory text to QoD and it isn't clear.

I'm happy to create a PR and discuss further.

### LINKED ISSUE #190: ICM text should not state that `Telco Operators` are the ones providing these APIs
**Problem description**

Assuming that only "Telco Operators" will implement these APIs is not valid, since MSO and other none "Telco" operators will implement many of these APIs.

Per commonalities [reduce telco-specific terms](https://github.com/camaraproject/Commonalities/blob/main/documentation/API-design-guidelines.md#25-reduce-telco-specific-terminology-in-api-definitions)

Additional references to telcos in the documentation should also be removed
 
**Expected behavior**

Use network operator or service provider rather than telco / telco operator

**Alternative solution**

**Additional context**

This will cause confusion with the mandatory text, since MSOs who are not Telco's are planning to deploy and are contributing to CAMARA

### LINKED ISSUE #200: Generalize documents by using "API provider" rather than "Telco Operator"
**Problem description**
Several text refer to the Telco / Operator as the API provider, hence excluding Channel partners.
Also the definition of the term User seems strange - is it needed in addition to "End-User".

**Expected action**
Update these texts to generically refer to API provider. Pointers:
in the ICM document: [CAMARA-API-access-and-user-consent.md](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-API-access-and-user-consent.md?plain=1#camara-api-specification---authorization-and-authentication-common-guidelines) 
- change the occurence of "Telco Operator" in the template text line 360 to "API Provider". - NOTE: this impacts all API specifications  referencing the template text. There is already an issue for that #190 which shall be done before chaning all the APIs.
- Also check and generalize as appropriate the other occurences of "Telco" or "operator" in the document (total 12 Telco and 81 operator).
- the definitions of User as an "operator subscriber" and "End-user" as the application user seems confusing. do we need to distinguish between an "End-User" of the application and a Telco subscriber "User" ? do we need User at all ? 
- line 365: avoid the word "customer": API customer -> API consumer

Also generalize "Telco" occurence in the ICM document: https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-Security-Interoperability.md

### PR DESCRIPTION
**Pull request by jpengar**
#### What type of PR is this?

* documentation

#### What this PR does / why we need it:

As discussed by the team in [11/09 WG call](https://lf-camaraproject.atlassian.net/l/cp/xHd81isJ), in order to move things forward, we are going to split the content of PR #182 into three different PRs:

* One focused on just updating the terms and definitions. So that the WG can agree on the terms to be used in the rest of the documentation. Initially it keeps original changes proposed in #182.
  * https://github.com/camaraproject/IdentityAndConsentManagement/pull/212
* Another editorial one, simply including the same "better English" changes proposed in #182. It assumes for the moment that the terms and definitions are the ones initially proposed. Then the document may need some adjustments later if something is changed in the previous PR and this one is already merged. This "better English" one will probably be the easiest to approve.
  * https://github.com/camaraproject/IdentityAndConsentManagement/pull/213
* ❗**THIS PR**❗And finally, another independent PR just for changes in the `info.description` template, which could also fix other issues like https://github.com/camaraproject/IdentityAndConsentManagement/issues/178.
  * https://github.com/camaraproject/IdentityAndConsentManagement/pull/214

The WG concluded that it would be much easier to deal with if we separated the proposed changes in #182 into different PRs.

#### Which issue(s) this PR fixes:

Fixes #178
Fixes #190 
Fixes #200  

#### Special notes for reviewers:

Further context in https://github.com/camaraproject/IdentityAndConsentManagement/pull/182

CC @chrishowell 

#### Changelog input

```
Improved document writing

```

#### Additional documentation 

N/A

### CODE REVIEW

**Comment by jpengar** [comment]
@chrishowell's suggestions are fine with me. But in this case, as we are updating the `info.description` template, which affects all API specs, I think the final agreed text will need to be validated by the TSC for final approval.

**Comment by jpengar** [comment]
I invite WG participants to review and comment. Specifically related issue owners. @RandyLevensalor
CC @AxelNennker @sebdewet

File: documentation/CAMARA-API-access-and-user-consent.md
Comment (RandyLevensalor) [inline documentation/CAMARA-API-access-and-user-consent.md:359]:
```suggestion
The specific authorization flows to be used will be determined during the onboarding process, happening between the API client and the operator exposing the API, taking into account the declared purpose for accessing the API, whilst also being subject to the prevailing legal framework dictated by local legislation.
```

Can we drop telco, since these APIs can be provided by MSOs and other non-telco operators?

Also the missing "the" was in my initial comment.

**Review by RandyLevensalor — CHANGES_REQUESTED** [review]
This is an improvement. Just a few minor comments.

**Comment by jpengar** [comment]
@camaraproject/tsc-participants As we are updating the `info.description` template, which affects all API specs, I think the final agreed text will need to be validated by the TSC for final approval as done in the past. Am I right?

File: documentation/CAMARA-API-access-and-user-consent.md
Comment (tanjadegroot) [inline documentation/CAMARA-API-access-and-user-consent.md:357]: replace "client" by "API consumer"

Can this text be more explicit and say "how an API consumer SHALL request an access token" ?

Comment (jpengar) [reply documentation/CAMARA-API-access-and-user-consent.md]: Fixed in https://github.com/camaraproject/IdentityAndConsentManagement/pull/214/commits/00bac931568697c72ba5cbfcfea76e83fffc8329. I've restored the link to the ICM WG, which I missed. I prefer to keep it as well. It was a change originally requested in #182.

File: documentation/CAMARA-API-access-and-user-consent.md
Comment (jpengar) [reply documentation/CAMARA-API-access-and-user-consent.md]: Fixed in https://github.com/camaraproject/IdentityAndConsentManagement/pull/214/commits/00bac931568697c72ba5cbfcfea76e83fffc8329

**Review by RandyLevensalor — APPROVED** [review]
This is much more clear. Thanks for making the improvements.

File: documentation/CAMARA-API-access-and-user-consent.md
Comment (sebdewet) [inline documentation/CAMARA-API-access-and-user-consent.md:345]:
```suggestion
The specific authorization flows to be used will be agreed upon during the onboarding process, happening between the provider of the application consuming the API and the operator's API exposure platform, taking into account the declared purpose for accessing the API, whilst also being subject to the prevailing legal framework dictated by local legislation.
```

**Review by eric-murray — APPROVED** [review]

**Comment by jpengar** [comment]
As discussed on yesterday's ICM call, please @sebdewet or @AxelNennker, we require a code owner (different from me 🙉) for final approval before proceeding with the merge of the PR. Could you kindly assist with this?

**Review by sebdewet — APPROVED** [review]

**Pull request merged**
